contracts/governance/Vesting/VestingRegistryLogic.sol

```diff
@@ -0,0 +1,292 @@
+pragma solidity ^0.5.17;
+pragma experimental ABIEncoderV2;
+
+import "../../interfaces/IERC20.sol";
+import "../IFeeSharingProxy.sol";
+import "./IVesting.sol";
+import "./ITeamVesting.sol";
+import "./VestingRegistryStorage.sol";
+import "../../openzeppelin/Initializable.sol";
+
+contract VestingRegistryLogic is VestingRegistryStorage, Initializable {
+	event SOVTransferred(address indexed receiver, uint256 amount);
+	event VestingCreated(
+		address indexed tokenOwner,
+		address vesting,
+		uint256 cliff,
+		uint256 duration,
+		uint256 amount,
+		uint256 vestingCreationType
+	);
+	event TeamVestingCreated(
+		address indexed tokenOwner,
+		address vesting,
+		uint256 cliff,
+		uint256 duration,
+		uint256 amount,
+		uint256 vestingCreationType
+	);
+	event TokensStaked(address indexed vesting, uint256 amount);
+
+	/**
+	 * @notice Replace constructor with initialize function for Upgradable Contracts
+	 * This function will be called only once by the owner
+	 * */
+	function initialize(
+		address _vestingFactory,
+		address _SOV,
+		address _staking,
+		address _feeSharingProxy,
+		address _vestingOwner,
+		address _lockedSOV,
+		address[] memory _vestingRegistries
+	) public onlyOwner initializer {
+		require(_SOV != address(0), "SOV address invalid");
+		require(_staking != address(0), "staking address invalid");
+		require(_feeSharingProxy != address(0), "feeSharingProxy address invalid");
+		require(_vestingOwner != address(0), "vestingOwner address invalid");
+		require(_lockedSOV != address(0), "LockedSOV address invalid");
+
+		_setVestingFactory(_vestingFactory);
+		SOV = _SOV;
+		staking = _staking;
+		feeSharingProxy = _feeSharingProxy;
+		vestingOwner = _vestingOwner;
+		lockedSOV = LockedSOV(_lockedSOV);
+		for (uint256 i = 0; i < _vestingRegistries.length; i++) {
+			require(_vestingRegistries[i] != address(0), "Vesting registry address invalid");
+			vestingRegistries.push(IVestingRegistry(_vestingRegistries[i]));
+		}
+	}
+
+	/**
+	 * @notice sets vesting factory address
+	 * @param _vestingFactory the address of vesting factory contract
+	 */
+	function setVestingFactory(address _vestingFactory) public onlyOwner {
+		_setVestingFactory(_vestingFactory);
+	}
+
+	/**
+	 * @notice Internal function that sets vesting factory address
+	 * @param _vestingFactory the address of vesting factory contract
+	 */
+	function _setVestingFactory(address _vestingFactory) internal {
+		require(_vestingFactory != address(0), "vestingFactory address invalid");
+		vestingFactory = IVestingFactory(_vestingFactory);
+	}
+
+	/**
+	 * @notice transfers SOV tokens to given address
+	 * @param _receiver the address of the SOV receiver
+	 * @param _amount the amount to be transferred
+	 */
+	function transferSOV(address _receiver, uint256 _amount) public onlyOwner {
+		require(_receiver != address(0), "receiver address invalid");
+		require(_amount != 0, "amount invalid");
```

Comment:
I think we cannot remove the check receiver != address(0) as it will fail with 'function call to a non-contract' while passing zero. I think it is good to have both the checks.